test/test_kfac.py

```diff
@@ -340,23 +423,37 @@ def test_kfac_mc_one_datum(
     report_nonclose(ggn, kfac_mat, rtol=rtol, atol=atol)
 
 
+@mark.parametrize(
+    "separate_weight_and_bias", [True, False], ids=["separate_bias", "joint_bias"]
+)
+@mark.parametrize(
+    "exclude", [None, "weight", "bias"], ids=["all", "no_weights", "no_biases"]
+)
 def test_kfac_ef_one_datum(
     kfac_exact_one_datum_case: Tuple[
         Module,
         Union[BCEWithLogitsLoss, CrossEntropyLoss],
         List[Parameter],
         Iterable[Tuple[Tensor, Tensor]],
     ],
+    separate_weight_and_bias: bool,
+    exclude: str,
 ):
```

Comment:
I wonder if the functionality inside this test can be refactored into a single function that is shared by all tests. This would reduce a lot of code duplication, e.g. `if exclude is not None: ...` condition.

Reply:
This is a bit of a tangent to the current PR, but since we are introducing a lot of new complexity, it might be worth reduce it elsewhere.

Reply:
Yes, I agree that there is tons of code duplication in the tests that could be reduced. I can maybe do some simple refactoring by just moving highly duplicated code into separate functions, as suggested, but leave more comprehensive refactoring for later.

Reply:
See [this commit](https://github.com/f-dangel/curvlinops/pull/173/commits/183d06feaed907e3c8a1a464608c3f99f8d2580d).